receiver/statsdreceiver/reporter.go

```diff
@@ -30,15 +40,89 @@ func newReporter(set receiver.CreateSettings) (transport.Reporter, error) {
 	if err != nil {
 		return nil, err
 	}
-	return &reporter{
+
+	r := &reporter{
 		logger:        set.Logger,
 		sugaredLogger: set.Logger.Sugar(),
 		obsrecv:       obsrecv,
-	}, nil
+		staticAttrs: []attribute.KeyValue{
+			attribute.String("receiver", set.ID.String()),
+		},
+	}
+
+	// See https://github.com/open-telemetry/opentelemetry-collector/blob/241334609fc47927b4a8533dfca28e0f65dad9fe/receiver/receiverhelper/obsreport.go#L104
+	// for the metric naming conventions
+
+	r.acceptedMetricPoints, err = metadata.Meter(set.TelemetrySettings).Int64Counter(
+		"receiver/accepted_metric_points",
+		metric.WithDescription("Number of metric data points accepted"),
+		metric.WithUnit("1"),
+	)
+	if err != nil {
+		return nil, err
+	}
+
+	r.refusedMetricPoints, err = metadata.Meter(set.TelemetrySettings).Int64Counter(
+		"receiver/refused_metric_points",
+		metric.WithDescription("Number of metric data points refused"),
+		metric.WithUnit("1"),
+	)
+	if err != nil {
+		return nil, err
+	}
+
+	r.flushedMetricPoints, err = metadata.Meter(set.TelemetrySettings).Int64Counter(
+		"receiver/statsd.flushed_metric_points",
+		metric.WithDescription("Number of metric data points flushed"),
+		metric.WithUnit("1"),
+	)
+	if err != nil {
+		return nil, err
+	}
+
+	r.flushCount, err = metadata.Meter(set.TelemetrySettings).Int64Counter(
+		"receiver/statsd.flushes",
+		metric.WithDescription("Number of flush operations"),
+		metric.WithUnit("1"),
+	)
+	if err != nil {
+		return nil, err
+	}
+
+	return r, nil
 }
 
 func (r *reporter) OnDebugf(template string, args ...any) {
 	if r.logger.Check(zap.DebugLevel, "debug") != nil {
 		r.sugaredLogger.Debugf(template, args...)
 	}
 }
+
+func (r *reporter) RecordReceivedMetric(err error) {
+	if err == nil {
+		r.acceptedMetricPoints.Add(context.Background(), 1, metric.WithAttributes(r.staticAttrs...))
+	} else {
+		r.refusedMetricPoints.Add(context.Background(), 1, metric.WithAttributes(r.staticAttrs...))
+	}
+}
+
+func (r *reporter) RecordFlushedMetrics(count int64, err error) {
```

Comment:
again compared to the prometheusreceiver, i think the flushed metrics should be the receiver accepted/refused metrics. The parser should be the place where the instrumentation for parse error counts are added

Reply:
Hmm.  I'm not sure about this one.  In our monitoring, we are comparing the received datapoint count to the sent metrics count from statsd clients.  See https://github.com/DataDog/datadog-go/blob/9d920252c04efe4e5f4d3a283002b72253a7da47/statsd/telemetry.go#L271

Reply:
I'm OK with flipping the logic if that's the consensus.  The naming becomes a bit harder though 🙂